src/components/Class/ClassForm.js

```diff
@@ -0,0 +1,720 @@
+            )}
+            <TextField
+              sx={{ mt: 3 }}
+              type="date"
+              variant="outlined"
+              inputProps={{
+                min: classFields?.date,
+              }}
+              value={classFields.date}
+              name="date"
+              label="Start Date"
+              fullWidth
+              onChange={(e) => {
+                changeHandler(e);
+              }}
+            />
+            {classFields.type === "batch" && (
+              <>
+                <FormLabel component="legend">
+                  <Typography
+                    variant="body2"
+                    color="text.secondary"
+                    sx={{ pt: 3 }}
+                    mb={2}
+                  >
+                    Schedule on days
+                  </Typography>
+                </FormLabel>
+                <FormGroup aria-label="position" row>
+                  {Object.keys(days).map((item) => (
+                    <FormControlLabel
+                      control={
+                        <Checkbox
+                          value={item}
+                          checked={classFields.on_days.includes(item)}
+                          onChange={handleDaySelection}
+                        />
+                      }
+                      label={item}
+                      labelPlacement={item}
+                    />
+                  ))}
+                </FormGroup>
+              </>
+            )}
+            <Grid container mt={2} spacing={2}>
+              <Grid item xs={isActive ? 12 : 6}>
+                <LocalizationProvider dateAdapter={AdapterDateFns}>
+                  <Stack spacing={3}>
+                    <DesktopTimePicker
+                      label="Start Time"
+                      value={classFields.start_time}
+                      onChange={(startTime) => {
+                        setClassFields({
+                          ...classFields,
+                          ["start_time"]: startTime,
+                        });
+                      }}
+                      renderInput={(params) => <TextField {...params} />}
+                    />
+                  </Stack>
+                </LocalizationProvider>
+              </Grid>
+              <Grid item xs={isActive ? 12 : 6}>
+                <LocalizationProvider dateAdapter={AdapterDateFns}>
+                  <Stack spacing={3}>
+                    <DesktopTimePicker
+                      label="End-Time"
+                      value={classFields.end_time}
+                      onChange={(endTime) => {
+                        setClassFields({
+                          ...classFields,
+                          ["end_time"]: endTime,
+                        });
+                      }}
+                      renderInput={(params) => <TextField {...params} />}
+                    />
+                  </Stack>
+                </LocalizationProvider>
+              </Grid>
```

Comment:
Not sure that it matters too much here, but maybe remove the duplication:
```suggestion
              {[
                { label: 'Start Time', prop: 'start_time' },
                { label: 'End Time', prop: 'end_time' },
              ].map(({label, prop}) => (
                <Grid item xs={isActive ? 12 : 6}>
                  <LocalizationProvider dateAdapter={AdapterDateFns}>
                    <Stack spacing={3}>
                      <DesktopTimePicker
                        label={label}
                        value={classFields[prop]}
                        onChange={(time) => {
                          setClassFields({
                            ...classFields,
                            [prop]: time,
                          });
                        }}
                        renderInput={(params) => <TextField {...params} />}
                      />
                    </Stack>
                  </LocalizationProvider>
                </Grid>
              ))}
```